spark/src/main/scala/org/apache/spark/sql/comet/CometBroadcastExchangeExec.scala

```diff
@@ -191,7 +193,7 @@ case class CometBroadcastExchangeExec(originalPlan: SparkPlan, child: SparkPlan)
   override protected def doExecuteColumnar(): RDD[ColumnarBatch] = {
     val broadcasted = executeBroadcast[Array[ChunkedByteBuffer]]()
 
-    new CometBatchRDD(sparkContext, broadcasted.value.length, broadcasted)
+    new CometBatchRDD(sparkContext, childRDD.getNumPartitions, broadcasted)
```

Comment:
This fixes a bug. The broadcast RDD must have same number of partitions as child RDD.

Reply:
Update. Child RDD partition number may also not be same as the zipping side. We need to get the number of partition of zipping side when triggering this execution method.

Reply:
This issue is described in #243.